### PR TITLE
BAI-1286 TokenAgent Improvements

### PR DESCRIPTION
**Pull request by bw27492**
- Now reads env variables or cli input

### CODE REVIEW

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: lib/python/src/bailo/core/agent.py [hunk 1]
```diff
@@ -95,14 +97,30 @@ def delete(self, *args, **kwargs):
...
         """
+        super().__init__()
+
+        if access_key is None or secret_key is None:
+            try:
+                access_key = os.environ["BAILO_ACCESS_KEY"]
+                secret_key = os.environ["BAILO_SECRET_KEY"]
+            except:
+                access_key = getpass.getpass("BAILO ACCESS KEY ")
+                secret_key = getpass.getpass("BAILO SECRET KEY ")
```
Comment (GB27247) [inline lib/python/src/bailo/core/agent.py:118]: Just be aware this input will not work for some environments. Such as emulators or pseudo-terminals.

File: lib/python/src/bailo/core/agent.py [hunk 2]
```diff
@@ -95,14 +97,30 @@ def delete(self, *args, **kwargs):
...
         :param secret_key: Secret key
+        :param save_tokens: Save tokens as environment variables, defaults to True
         """
+        super().__init__()
+
+        if access_key is None or secret_key is None:
+            try:
+                access_key = os.environ["BAILO_ACCESS_KEY"]
+                secret_key = os.environ["BAILO_SECRET_KEY"]
+            except:
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/core/agent.py:116]: Done

File: lib/python/src/bailo/core/agent.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/core/agent.py:118]: Done

[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/core/agent.py:118]: Done

File: lib/python/src/bailo/core/agent.py [hunk 3]
```diff
@@ -95,14 +97,30 @@ def delete(self, *args, **kwargs):
...
+                access_key = getpass.getpass("BAILO ACCESS KEY ")
+                secret_key = getpass.getpass("BAILO SECRET KEY ")
+
+        if save_tokens:
+            os.environ["BAILO_ACCESS_KEY"] = access_key
+            os.environ["BAILO_SECRET_KEY"] = secret_key
+
         self.access_key = access_key
         self.secret_key = secret_key
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/core/agent.py:125]: Now checks when instantiating a client object (generally for all auth methods)

File: lib/python/src/bailo/core/agent.py
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/core/agent.py:1]: No longer needed as all endpoints can take tokens

**Review by GB27247 — APPROVED** [review]

**Pull request merged**